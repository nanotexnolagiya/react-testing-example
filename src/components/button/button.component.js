import './button.style.css'

/**
 * Custom button component
 *
 * @component
 * @example
 * const className = 'custom-button-selector'
 * const disabled = false
 * return (
 *   <Button className={className} disabled={disabled} />
 * )
 */
export const Button = ({ className, ...props }) => {
  return <button data-testid="custom-button" className={'form-button ' + (className || '')} {...props} />
}