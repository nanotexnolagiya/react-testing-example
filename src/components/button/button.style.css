.form-button {
  outline: none;
  padding: 10px 15px;
  border: 1px solid #111111;
  border-radius: 5px;
  background-color: transparent;
  cursor: pointer;
}