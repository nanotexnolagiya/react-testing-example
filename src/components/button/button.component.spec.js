import { render, screen } from "@testing-library/react";
import { Button } from "./button.component";


it('Button component should be render', () => {
  const { asFragment } = render(<Button />);

  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('custom-button')).toBeInTheDocument();
})