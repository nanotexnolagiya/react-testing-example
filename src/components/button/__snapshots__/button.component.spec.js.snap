// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should be render 1`] = `
<DocumentFragment>
  <button
    class="form-button "
    data-testid="custom-button"
  />
</DocumentFragment>
`;
